packages/core/src/operator/getChallengerPublicKey.ts

```diff
@@ -0,0 +1,20 @@
+import { ethers } from "ethers";
+import { config } from "../config.js";
+import { getProvider, RefereeAbi } from "../index.js";
+
+
+
+export async function getChallengerPublicKey(): Promise<string> {
+
+    try {
+        const provider = getProvider();
+        const referee = new ethers.Contract(config.refereeAddress, RefereeAbi, provider);
+        const publicKey = await referee.challengerPublicKey();
+
+        return publicKey;
+        
+    } catch (error: unknown) {
+        const message = error instanceof Error ? error.message : 'An unknown error occurred';
+        throw new Error(`Error getting challenger public key: ${message}`);        
+    }
```

Comment:
Thsi try catch is only restructuring the error which results in loosing the original stack trace.
The service function is not expected to handle errors, this should be handled by the caller.

Reply:
Removed, thank you